pkg/networkservice/common/refresh/client.go

```diff
@@ -59,15 +61,24 @@ func (t *refreshClient) Request(ctx context.Context, request *networkservice.Net
 	// Set its connection to the returned connection we received
 	req.Connection = rv
 	// Setup the timer with the req containing the returned connection
-	ct, err := t.createTimer(ctx, req, opts...)
+
+	expire, err := t.getExpireDuration(request)
 	if err != nil {
 		return nil, errors.Wrapf(err, "Error creating timer from Request.Connection.Path.PathSegment[%d].ExpireTime", request.GetConnection().GetPath().GetIndex())
 	}
 	t.executor.AsyncExec(func() {
+		if ctx.Err() != nil {
```

Comment:
Why do we need to check request context here?

Reply:
For each scheduled refresh request (created in `createTimer`) we set cancel function and save it to the map. When a new request from heal or stop call with the same `connId` will reach `AsyncExec` block, it cancels refresh's context. Then when refresh will try to schedule its next call, it sees that context was cancelled and terminates itself. This check must be placed in `AsyncExec` block to ensure that new request and refresh goroutines will not get access to context concurrently.

Reply:
As I can see here using the request context. Should to use a special `refresh context` for refresh goals?
For example you need to create a refreshing context on first Request call and reuse this context in the refresh timer. Is it make sense?